ram.tf

```diff
@@ -1,3 +1,15 @@
+locals {
+  ram_principals_provided = var.ram_principal != null || length(var.ram_principals) > 0
+  ram_principals = toset(var.ram_resource_share_enabled ? toset(
+    local.ram_principals_provided ? concat(
+      var.ram_principal == null ? [] : [var.ram_principal],
+      var.ram_principals,
+      ) : [
+      data.aws_organizations_organization.default[0].arn
```

Comment:
will it work when `var.ram_resource_share_enabled = false`?

Usually we use 

```
join(data.aws_organizations_organization.default.*.arn, "")

```

which works in all cases